homeassistant/components/watergate/sensor.py

```diff
@@ -0,0 +1,257 @@
+        if data.state and data.state.water_meter
+        else None,
+        translation_key="water_meter_volume",
+        key="water_meter_volume",
+        native_unit_of_measurement=UnitOfVolume.LITERS,
+        device_class=SensorDeviceClass.WATER,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.state.water_meter.duration
+        if data.state and data.state.water_meter
+        else None,
+        translation_key="water_meter_duration",
+        key="water_meter_duration",
+        native_unit_of_measurement=UnitOfTime.MINUTES,
+        device_class=SensorDeviceClass.DURATION,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.networking.ip if data.networking else None,
+        translation_key="ip",
+        key="ip",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.networking.gateway if data.networking else None,
+        translation_key="gateway",
+        key="gateway",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.networking.subnet if data.networking else None,
+        translation_key="subnet",
+        key="subnet",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.networking.ssid if data.networking else None,
+        translation_key="ssid",
+        key="ssid",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.networking.rssi if data.networking else None,
+        translation_key="rssi",
+        key="rssi",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.networking.wifi_uptime if data.networking else None,
+        translation_key="wifi_uptime",
+        key="wifi_uptime",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement=UnitOfTime.MILLISECONDS,
+        device_class=SensorDeviceClass.DURATION,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.networking.mqtt_uptime if data.networking else None,
+        translation_key="mqtt_uptime",
```

Comment:
same for all uptime sensors

Reply:
note: we do not allow "uptime" sensors.
This should be converted to a "up since" sensor with a valid datetime.
The same applies to wifi (and maybe others) uptime sensors.

Reply:
@epenet should I add another platform then? 'datetime' platform? Or just convert it in the function to datetime and keep it in this sensor?